kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/duplicates.kt

```diff
@@ -6,48 +6,130 @@ import io.kotest.matchers.MatcherResult
 import io.kotest.matchers.should
 import io.kotest.matchers.shouldNot
 
-fun <T, I : Iterable<T>> I.shouldContainDuplicates(): I {
-   toList().shouldContainDuplicates()
+fun BooleanArray.shouldContainDuplicates(): BooleanArray {
+   asList() should containDuplicates("BooleanArray")
    return this
 }
 
-fun <T> Array<T>.shouldContainDuplicates() {
-   asList().shouldContainDuplicates()
+fun BooleanArray.shouldNotContainDuplicates(): BooleanArray {
+   asList() shouldNot containDuplicates("BooleanArray")
+   return this
 }
 
-fun <T, C : Collection<T>> C.shouldContainDuplicates(): C {
-   this should containDuplicates()
+fun ByteArray.shouldContainDuplicates(): ByteArray {
+   asList() should containDuplicates("ByteArray")
    return this
 }
 
-fun <T, I : Iterable<T>> I.shouldNotContainDuplicates(): I {
-   toList().shouldNotContainDuplicates()
+fun ByteArray.shouldNotContainDuplicates(): ByteArray {
+   asList() shouldNot containDuplicates("ByteArray")
+   return this
+}
+
+fun ShortArray.shouldContainDuplicates(): ShortArray {
+   asList() should containDuplicates("ShortArray")
+   return this
+}
+
+fun ShortArray.shouldNotContainDuplicates(): ShortArray {
+   asList() shouldNot containDuplicates("ShortArray")
+   return this
+}
+
+fun CharArray.shouldContainDuplicates(): CharArray {
+   asList() should containDuplicates("CharArray")
+   return this
+}
+
+fun CharArray.shouldNotContainDuplicates(): CharArray {
+   asList() shouldNot containDuplicates("CharArray")
+   return this
+}
+
+fun IntArray.shouldContainDuplicates(): IntArray {
+   asList() should containDuplicates("IntArray")
+   return this
+}
+
+fun IntArray.shouldNotContainDuplicates(): IntArray {
+   asList() shouldNot containDuplicates("IntArray")
+   return this
+}
+
+fun LongArray.shouldContainDuplicates(): LongArray {
+   asList() should containDuplicates("LongArray")
+   return this
+}
+
+fun LongArray.shouldNotContainDuplicates(): LongArray {
+   asList() shouldNot containDuplicates("LongArray")
+   return this
+}
+
+fun FloatArray.shouldContainDuplicates(): FloatArray {
+   asList() should containDuplicates("FloatArray")
+   return this
+}
+
+fun FloatArray.shouldNotContainDuplicates(): FloatArray {
+   asList() shouldNot containDuplicates("FloatArray")
+   return this
+}
+
+fun DoubleArray.shouldContainDuplicates(): DoubleArray {
+   asList() should containDuplicates("DoubleArray")
+   return this
+}
+
+fun DoubleArray.shouldNotContainDuplicates(): DoubleArray {
+   asList() shouldNot containDuplicates("DoubleArray")
+   return this
+}
+
+fun <T> Array<T>.shouldContainDuplicates(): Array<T> {
+   asList() should containDuplicates("Array")
    return this
 }
 
 fun <T> Array<T>.shouldNotContainDuplicates(): Array<T> {
-   asList().shouldNotContainDuplicates()
+   asList() shouldNot containDuplicates("Array")
+   return this
+}
+
+fun <T, C : Collection<T>> C.shouldContainDuplicates(): C {
```

Comment:
If we have Iterable, do we need the collection overflow still?

Reply:
No, it is scheduled for removal in a followup PR. Alex pointed out that these removals should be added as separate PRs, which I'll do. In fact, the removal of this Collection overload is already waiting for this PR to be merged - #4371